src/cli/Peers.ts

```diff
@@ -0,0 +1,387 @@
+import fs from 'fs';
+import commander from 'commander';
+import { PolykeyAgent, PeerInfo } from '../lib/Polykey';
+import {} from '../../tmp/hole-punch-test/publicserver';
+import { actionRunner, pkLogger, PKMessageType, determineNodePath } from '.';
+
+function makeAddPeerCommand() {
+  return new commander.Command('add')
```

Comment:
I would separate the command creation from the handler declaration. Sort of like router vs controllers/resources as in HTTP. See loja-api to see what I'm talking about.

Reply:
I am not sure I follow, as far as I can tell you mean something like this?
```
function makeAddPeerCommand() {
  return new commander.Command('add')
    .description('add a new peer to the store')
}
function addHandlersToAddPeerCommand(command: Command) {
  return command
    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
    .action(...)
}
function makePeersCommand() {
  return new commander.Command('peers')
    .description('peer operations')
    .addCommand(addHandlersToAddPeerCommand(makeAddPeerCommand()))
}
```

It seems like it complicates things, the handlers are linked only to one command and its kind of a self contained unit.